### PR TITLE
Data structure stack

### PR DESCRIPTION
**Pull request by DavidMejia025**

### CODE REVIEW

File: graphs_algorithms_and_data_structures/some_basic_data_structures/linked_list.rb [hunk 1]
```diff
@@ -0,0 +1,159 @@
+# LinkedList data structucture that can be seen as a unidirectional graph where
+# each  vertex can have maximum two connections, and all the nodes are
+# connected "in-line" .
+class LinkedList
+  def initialize
+    @root    = nil
+    @current = nil
+  end
+
+  def current
```
Comment (germanescobar) [inline graphs_algorithms_and_data_structures/some_basic_data_structures/linked_list.rb:10]: Don't leak the implementation

File: graphs_algorithms_and_data_structures/some_basic_data_structures/linked_list.rb [hunk 2]
```diff
@@ -0,0 +1,159 @@
+# LinkedList data structucture that can be seen as a unidirectional graph where
+# each  vertex can have maximum two connections, and all the nodes are
+# connected "in-line" .
+class LinkedList
+  def initialize
+    @root    = nil
+    @current = nil
```
Comment (germanescobar) [inline graphs_algorithms_and_data_structures/some_basic_data_structures/linked_list.rb:7]: Having a `@current` is an optimization but I would prefer that you do it only with the `@root`

File: graphs_algorithms_and_data_structures/some_basic_data_structures/linked_list.rb [hunk 3]
```diff
@@ -0,0 +1,159 @@
...
+  end
+
+  def current
+    @current
+  end
+
+  def root
```
Comment (germanescobar) [inline graphs_algorithms_and_data_structures/some_basic_data_structures/linked_list.rb]: Don't leak the implementation.

File: graphs_algorithms_and_data_structures/some_basic_data_structures/queue.rb [hunk 4]
```diff
@@ -0,0 +1,63 @@
...
+
+  def show
+    @queue
+  end
+
+  def add(element)
+    @queue.push(element)
+  end
+
+  def remove
```
Comment (germanescobar) [inline graphs_algorithms_and_data_structures/some_basic_data_structures/queue.rb]: A better name would be `dequeue`

**Review by germanescobar — APPROVED** [review]
Awesome job @DavidMejia025! I think you grabbed the concept of the nodes already 💪

File: graphs_algorithms_and_data_structures/some_basic_data_structures/queue.rb (hunk 4 above)
Comment (DavidMejia025) [reply graphs_algorithms_and_data_structures/some_basic_data_structures/queue.rb]: @germanescobar  Done

File: graphs_algorithms_and_data_structures/some_basic_data_structures/linked_list.rb (hunk 2 above)
Comment (DavidMejia025) [reply graphs_algorithms_and_data_structures/some_basic_data_structures/linked_list.rb:7]: @germanescobar  I already add another implementation getting rid of current variable. Please take a look when you can. I still missing how to deal without leaking the root variable.

Thinking out loud  Maybe linked list should only be (@val and @ref) like the basic node  but I will think about that tomorrow.

**Pull request merged**